lib/lhm/migrator.rb

```diff
@@ -188,7 +189,11 @@ def validate
       end
 
       unless @origin.satisfies_id_autoincrement_requirement?
```

Comment:
There's no way for this to not error out when an `order_column` is specified.

`@origin.satisfies_id_autoincrement_requirement?` checks the `id` column, even if an `order_column` is set. Once that check fails, an error is set no matter what. The code should probably look more like:

```
      if @options[:order_column]
        if !@origin.can_use_order_column?(@options[:order_column])
          error("origin does not satisfy primary key requirements")
        end
      else
        if !@origin.satisfies_id_autoincrement_requirement?
          error("origin does not satisfy primary key requirements")
        end
      end
```

Sorry for the comment, not sure what the best way to submit changes to a PR is. Thanks!